atom/common/api/atom_bindings.cc

```diff
@@ -238,4 +253,53 @@ v8::Local<v8::Value> AtomBindings::GetIOCounters(v8::Isolate* isolate) {
   return dict.GetHandle();
 }
 
+// static
+base::FilePath AtomBindings::TakeHeapSnapshot(v8::Isolate* isolate,
+                                              mate::Arguments* args) {
+  base::ScopedAllowBlockingForTesting allow_blocking;
+
+  auto file_path = HeapSnapshotOutputStream::GetFilePath();
+
+  base::File file(file_path,
+                  base::File::FLAG_CREATE_ALWAYS | base::File::FLAG_WRITE);
+
+  if (!file.IsValid()) {
+    auto* render_frame = GetCurrentRenderFrame();
+    if (!render_frame) {
+      args->ThrowError("takeHeapSnapshot failed");
+      return base::FilePath();
+    }
+
+    IPC::PlatformFileForTransit file_handle;
+    auto* message = new AtomFrameHostMsg_CreateHeapSnapshotFile(
+        render_frame->GetRoutingID(), &file_path, &file_handle);
+
+    if (!render_frame->Send(message)) {
+      args->ThrowError("takeHeapSnapshot failed");
+      return base::FilePath();
+    }
+
+    file = IPC::PlatformFileForTransitToFile(file_handle);
+  }
+
+  if (!file.IsValid()) {
+    args->ThrowError("takeHeapSnapshot failed - cannot create file");
```

Comment:
Can you please move all error messages to constants in an anonymous namespace on the top of the file?

Reply:
why? we're not doing that it other places